pygments/lexers/maxima.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+"""
+    pygments.lexers.maxima
+    ~~~~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for the computer algebra system Maxima.
+
+    Derived from pygments/lexers/algebra.py.
+"""
+
+import re
+
+from pygments.lexer import RegexLexer, bygroups, words
+from pygments.token import Text, Comment, Operator, Keyword, Name, String, \
+    Number, Punctuation
+
+__all__ = ['MaximaLexer']
+
+class MaximaLexer(RegexLexer):
+    """
+    A `Maxima <http://maxima.sourceforge.net>`_ lexer.
+    Derived from pygments.lexers.MuPADLexer.
+    """
+    name = 'Maxima'
+    aliases = ['maxima', 'macsyma']
+    filenames = ['*.mac', '*.max']
+
+    terminated = r'(?=[ "()\'\n,;`])'  # whitespace or terminating macro characters
+
+    tokens = {
+        'root': [
+            (r'/\*', Comment.Multiline, 'comment'),
+            (r'"(?:[^"\\]|\\.)*"', String),
+            (r'\(|\)|\[|\]|\{|\}', Punctuation),
+            (r'''(?x)\b(?:
+                if|then|else|elseif|
+                do|while|
+                repeat|until|
+                for|from|to|downto|step|thru
+              )\b''', Keyword),
+            (r'''(?x)\b(?:
+                %pi|%e|%phi|%gamma|%i|
+                und|ind|infinity|inf|minf|
+                true|false|unknown
+              )\b''',
+             Name.Constant),
+            (r'\.|:|=|#|\+|-|\*|/|\^|@|>|<|\||!|\'|~=', Operator),
+            (r'''(?x)\b(?:
+                and|or|not
+              )\b''',
+             Operator.Word),
+            (r'''(?x)
+              ((?:[a-zA-Z_#][\w#]*|`[^`]*`)
+              (?:::[a-zA-Z_#][\w#]*|`[^`]*`)*)(\s*)([(])''',
+             bygroups(Name.Function, Text, Punctuation)),
+            (r'''(?x)
+              (?:[a-zA-Z_#%][\w#%]*|`[^`]*`)
+              (?:::[a-zA-Z_#%][\w#%]*|`[^`]*`)*''', Name.Variable),
+            (r'[-+]?\d+\.?' + terminated, Number.Integer),
+            (r'[-+]?(\d*\.\d+([bdefls][-+]?\d+)?|\d+(\.\d*)?[bdefls][-+]?\d+)'
+                + terminated, Number.Float),
+            (r'.', Text)
+        ],
+        'comment': [
+            (r'[^*/]', Comment.Multiline),
+            (r'/\*', Comment.Multiline, '#push'),
+            (r'\*/', Comment.Multiline, '#pop'),
+            (r'[*/]', Comment.Multiline)
```

Comment:
This should be `+` so you don't create one token per character (see test output.)

Reply:
Hi @Anteru , I see what you mean about not creating one token per character, but I don't understand what you mean where you say "This should be +" -- what is it that should be `+`? Thanks for the clarification.

Reply:
The regex rule should have a `+` after it, so it will match multiple tokens. Right now `[^*/]` matches _one_ character that is not from the set `*`, `/`, and immediately emits a token. If you have a line with `ab` you'll thus get two tokens, one per character. If you convert it to `[^*/]`, `ab` gets matched as a single token `ab`.

Reply:
I've reworked the comment patterns so that the whole content of the comment is captured as one token.